core/src/Streamly/Internal/Data/Stream/Container.hs

```diff
@@ -47,12 +48,14 @@ import qualified Streamly.Internal.Data.Stream.Transformer as Stream
 
 #include "DocTestDataStream.hs"
 
--- | The memory used is proportional to the number of unique elements in the
--- stream. If we want to limit the memory we can just use "take" to limit the
--- uniq elements in the stream.
-{-# INLINE_NORMAL nub #-}
-nub :: (Monad m, Ord a) => Stream m a -> Stream m a
-nub (Stream step1 state1) = Stream step (Set.empty, state1)
+-- | Return a subsequence of the stream removing any duplicate elements.
+--
+-- The memory used is proportional to the number of unique elements in the
+-- stream. One way to limit the memory is to  use @take@ on the resulting
+-- stream to limit the unique elements in the stream.
+{-# INLINE_NORMAL ordNub #-}
+ordNub :: (Monad m, Ord a) => Stream m a -> Stream m a
+ordNub (Stream step1 state1) = Stream step (Set.empty, state1)
```

Comment:
Why ord prefix? Does it return an ordered stream or require an ordered stream? Can that be mentioned in the documentation?